test/test_sphere_original.py

```diff
@@ -0,0 +1,112 @@
+import unittest
+
+from src.sphere_original import Sphere
+
+
+class SphereOriginalTest(unittest.TestCase):
+    def test_sphere_x_returns_correct_x(self):
+        sphere = Sphere(0, 1, 2, 1)
+        self.assertEqual(0, sphere.x)
+
+    def test_sphere_y_returns_correct_x(self):
+        sphere = Sphere(0, 1, 2, 1)
+        self.assertEqual(1, sphere.y)
+
+    def test_sphere_z_returns_correct_x(self):
+        sphere = Sphere(0, 1, 2, 1)
+        self.assertEqual(2, sphere.z)
+
+    def test_sphere_radius_returns_correct_radius(self):
+        sphere = Sphere(0, 0, 0, 1)
+        self.assertEqual(1, sphere.radius)
+
+    def test_diameter_various_spheres_returns_correct_diameter(self):
+        cases = [
+            Sphere(0, 0, 0, 0),
+            Sphere(0, 0, 0, 1),
+            Sphere(1, 1, 1, 0),
+            Sphere(10, 11, 12, 10),
+        ]
+        expecteds = [0, 2, 0, 20]
+        for (case, expect) in zip(cases, expecteds):
+            with self.subTest(case=case, expect=expect):
+                self.assertAlmostEqual(expect, case.diameter(), 5)
+
+    def test_surface_area_various_spheres_returns_correct_surface_area(self):
+        cases = [
+            Sphere(0, 0, 0, 0),
+            Sphere(0, 0, 0, 1),
+            Sphere(1, 1, 1, 0),
+            Sphere(10, 11, 12, 10),
+        ]
+        expecteds = [0, 12.56637, 0, 1256.63706]
+        for (case, expect) in zip(cases, expecteds):
+            with self.subTest(case=case, expect=expect):
+                self.assertAlmostEqual(expect, case.surface_area(), 5)
+
+    def test_volume_various_spheres_returns_correct_volume(self):
+        cases = [
+            Sphere(0, 0, 0, 0),
+            Sphere(0, 0, 0, 1),
+            Sphere(1, 1, 1, 0),
+            Sphere(10, 11, 12, 10),
+        ]
+        expecteds = [0, 4.18879, 0, 4188.7902]
+        for (case, expect) in zip(cases, expecteds):
+            with self.subTest(case=case, expect=expect):
+                self.assertAlmostEqual(expect, case.volume(), 5)
+
+    def test_distance_between_centres_various_spheres_returns_correct_distance(self):
+        cases = [
+            (Sphere(0, 0, 0, 1), Sphere(0, 0, 0, 1)),
```

Comment:
I'd probably use a [named tuple](https://docs.python.org/3/library/collections.html#collections.namedtuple), or a map for this. Definitely clearer to say `assert case.this.distance(case.that) is whatever` imo.

Though, it's local enough that it's fine as is -- mostly taste.

Reply:
Given the coolness of this idea, I started to change it over, but opted to stop since, although this test class is more for _me_, it is nearly the same as the better `Sphere` class and adding more complexity there (new idea of namedtuple) may be a lot for the students at this stage.